glfw/src/buildfile

```diff
@@ -6,59 +6,42 @@ pub = [dir_path] ../include/GLFW/
 include $pub
 
 pub_hdrs = $pub/{$($pub/ pub_hdrs)}
-cmn_hdrs = h{internal mappings}
-cmn_srcs = context.c init.c \
-           input.c monitor.c \
-           vulkan.c window.c             
 
-lib{glfw}: $cmn_hdrs $pub_hdrs c{$cmn_srcs}: include = ($cocoa == false)
-
-# Macos
-# Using adhoc receipes to compile all sources as objective-C files
-# Extensions for files used in variable to allow wildcard filtering
-# Otherwise both header and source code is matched
+# Common sources and headers.
 #
-cocoa_files = cocoa_*.m nsgl_context.m $cmn_srcs \
-              posix_thread.c cocoa_time.c \
-              egl_context.c osmesa_context.c
-for cocoa_file: $cocoa_files
-{
-  n = $name($cocoa_file)
-  obja{"$n".a.o}: file{"$cocoa_file"}
-  {{
-     diag objective-c ($<[0])
-     $c.path $c.poptions $cc.poptions $c.coptions $cc.coptions\
-     -c $path($<[0]) -o $path($>)
-  }}
-  objs{"$n".dylib.o}: file{"$cocoa_file"}
-  {{
-     diag objective-c ($<[0])
-     $c.path $c.poptions $cc.poptions $c.coptions $cc.coptions\
-     -c $path($<[0]) -o $path($>)
-  }}
 
-  liba{glfw}: obja{"$n".a.o}: include = $cocoa
-  libs{glfw}: objs{"$n".dylib.o}: include = $cocoa
-}
+cmn_hdrs = h{internal platform mappings null_platform null_joystick}
+cmn_srcs = c{context init input monitor platform vulkan window egl_context osmesa_context null_init null_monitor null_window null_joystick}
+
+
+lib{glfw}: $cmn_hdrs $pub_hdrs $cmn_srcs
+
+
+# MacOS
+lib{glfw}: m{cocoa_joystick cocoa_monitor cocoa_init cocoa_window nsgl_context} \
+           c{posix_thread posix_module cocoa_time} \
+           h{cocoa_platform cocoa_joystick cocoa_time posix_thread}: include = $cocoa
 
-lib{glfw}: h{cocoa* nsgl_context osmesa_context\
-             egl_context posix_thread}: include = $cocoa
 
 # Windows
-lib{glfw}: {h c}{win32* ?gl_context osmesa_context}: include = $win32
+lib{glfw}: c{win32_joystick win32_monitor win32_window win32_init win32_module wgl_context} \
+           h{win32_joystick win32_platform win32_time win32_thread}: include = $win32
+
+
+# Unix
+lib{glfw}: c{posix_time posix_thread posix_module} h{posix_time posix_thread}: include = $unix
 
 # Unix - X11
-lib{glfw}: {h c}{x11_* xkb_unicode posix_* linux_joystick \
-                 glx_context egl_context osmesa_context}: include = $usex11
+lib{glfw}: c{x11_init x11_monitor x11_window glx_context} h{x11_platform}: include = $build_x11
 
 # Unix - Wayland
-lib{glfw}: {h c}{wl_* xkb_unicode posix_* linux_joystick \
-                 egl_context osmesa_context}: include = $usewayland
-
+lib{glfw}: c{wl_init wl_monitor wl_window} h{wl_platform}: include = $build_wayland
 
-# Unix - OSMESA
-lib{glfw}: {h c}{null_* posix_* osmesa_context}: include = $useosmesa
+# Unix - X11 and Wayland
```

Comment:
Question: If both backends are merged (and selected at runtime), perhaps we don't need to divert both flags and could group them into a single one, say, e.g., build_linux?

Reply:
GLFW allows disabling either backend, so I don't think restricting this makes sense